server/routes/api.route.js

```diff
@@ -18,106 +18,82 @@ apiRouter.get("/status", HttpControllerWrapper(ApiController.status));
 
 apiRouter.get("/question/:type", HttpControllerWrapper(QuestionController.generateQuestion));
 
-apiRouter.get("/users/", authenticationMiddleware, HttpControllerWrapper(UserController.getAll));
+apiRouter.get("/users/", AuthMiddleware(), HttpControllerWrapper(UserController.getAll));
 
-apiRouter.post(
-  "/users/",
-  authenticationMiddleware,
-  HttpControllerWrapper(UserController.severalGetInfos)
-);
+apiRouter.post("/users/", AuthMiddleware(), HttpControllerWrapper(UserController.severalGetInfos));
 
 apiRouter.post("/users/login", HttpControllerWrapper(UserController.login));
 
-apiRouter.get(
-  "/users/:pseudo",
-  authenticationMiddleware,
-  HttpControllerWrapper(UserController.getInfos)
-);
+apiRouter.post("/users/token", HttpControllerWrapper(UserController.generateAccessToken));
+
+apiRouter.post("/users/logout", AuthMiddleware(), HttpControllerWrapper(UserController.logout));
+
+apiRouter.get("/users/:pseudo", AuthMiddleware(), HttpControllerWrapper(UserController.getInfos));
 
 apiRouter.patch(
   "/users/:pseudo",
-  authenticationMiddleware,
+  AuthMiddleware(),
   HttpControllerWrapper(UserController.saveInfos)
 );
 
-apiRouter.post(
-  "/duels/new",
-  authenticationMiddleware,
-  HttpControllerWrapper(DuelController.create)
-);
+apiRouter.post("/duels/new", AuthMiddleware(), HttpControllerWrapper(DuelController.create));
 
-apiRouter.get("/duels/", authenticationMiddleware, HttpControllerWrapper(DuelController.fetchAll));
+apiRouter.get("/duels/", AuthMiddleware(), HttpControllerWrapper(DuelController.fetchAll));
 
-apiRouter.get("/duels/:id", authenticationMiddleware, HttpControllerWrapper(DuelController.fetch));
+apiRouter.get("/duels/:id", AuthMiddleware(), HttpControllerWrapper(DuelController.fetch));
 
-apiRouter.post(
-  "/duels/:id/:round",
-  authenticationMiddleware,
-  HttpControllerWrapper(DuelController.play)
-);
+apiRouter.post("/duels/:id/:round", AuthMiddleware(), HttpControllerWrapper(DuelController.play));
 
 const FILES_DIR = process.env.NODE_ENV === "test" ? "files-test" : "files";
 
-apiRouter.use(
-  "/files/molecules",
-  authenticationMiddleware,
-  express.static(`${FILES_DIR}/molecules`)
-);
+apiRouter.use("/files/molecules", AuthMiddleware(true), express.static(`${FILES_DIR}/molecules`));
 
-apiRouter.use("/files/users", authenticationMiddleware, express.static(`${FILES_DIR}/users`));
+apiRouter.use("/files/users", AuthMiddleware(true), express.static(`${FILES_DIR}/users`));
```

Comment:
I would define a `const ONLY_ADMINS = true` in this file to make the routes clearer

Reply:
yep good idea !